File: 2015/01/main.go
package main

import (
	"fmt"
	"os"
)

func Part1() int {
	data, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	floor := 0
	for _, character := range data {
		if character == '(' {
			floor++
		} else if character == ')' {
			floor--
		}
	}
	return floor
}

func Part2() int {
	data, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	floor := 0
	for index, character := range data {
		if character == '(' {
			floor++
		} else if character == ')' {
			floor--
		}
		if floor == -1 {
			return index + 1
		}
	}
	return -1
}

func main() {
	fmt.Println(Part1())
	fmt.Println(Part2())
}

File: 2015/01/main.py
def convert(c):
    if c == "(":
        return 1
    else:
        return -1


def Part1():
    with open("input.txt") as f:
        return sum([convert(d) for d in f.read()])


def Part2():
    with open("input.txt") as f:
        count = 0
        for i, d in enumerate(f.read()):
            count += convert(d)
            if count < 0:
                return i + 1


print(Part1())
print(Part2())
